src/stores/appcatalog/actions.ts

```diff
@@ -23,3 +29,66 @@ export const listCatalogs = createAsynchronousAction<any, any, any>({
   shouldPerform: () => true,
   throwOnError: false,
 });
+
+export const installLatestIngress = createAsynchronousAction<
+  IInstallIngress,
+  IState,
+  void
+>({
+  actionTypePrefix: 'INSTALL_INGRESS_APP',
+  perform: async (state, dispatch, payload) => {
+    if (payload?.clusterId) {
+      const gsCatalog = selectIngressCatalog(state);
+
+      const { name, version } = gsCatalog.apps[
+        Constants.INSTALL_INGRESS_TAB_APP_NAME
+      ][0];
+
+      const appToInstall = {
+        name,
+        chartName: name,
+        version,
+        catalog: Constants.INSTALL_INGRESS_TAB_APP_CATALOG_NAME,
+        namespace: 'kube-system',
+        valuesYAML: '',
+        secretsYAML: '',
+      };
+
+      /* eslint-disable @typescript-eslint/await-thenable */
+      await dispatch(
```

Comment:
any idea what to do here? `await` is required because the following action will fire immediately thus making it somewhat wrong here.